server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -500,6 +508,54 @@ private void startLookups(List<LookupBean> lookupBeanList)
     }
   }
 
+  /**
+   * @return a map with successful lookups
+   */
+  private Map<String, LookupExtractorFactoryContainer> startLookups(
+      List<LookupBean> lookupBeans,
+      CompletionService<Map.Entry<String, LookupExtractorFactoryContainer>> completionService,
+      List<Future<?>> allFutures
+  ) throws InterruptedException
+  {
+    for (LookupBean lookupBean : lookupBeans) {
+      allFutures.add(completionService.submit(() -> startLookup(lookupBean)));
+    }
+    Map<String, LookupExtractorFactoryContainer> successfulLookups = new HashMap<>();
+    for (int i = 0; i < lookupBeans.size(); i++) {
+      Future<Map.Entry<String, LookupExtractorFactoryContainer>> completedFuture = completionService.take();
+      try {
+        Map.Entry<String, LookupExtractorFactoryContainer> lookupResult = completedFuture.get();
+        if (lookupResult != null) {
+          successfulLookups.put(lookupResult.getKey(), lookupResult.getValue());
+        }
+      }
+      catch (ExecutionException e) {
+        LOG.error(e.getCause(), "Exception while starting a lookup");
+        // not adding to successfulLookups
+      }
+    }
+    return successfulLookups;
+  }
+
+  @Nullable
+  private Map.Entry<String, LookupExtractorFactoryContainer> startLookup(LookupBean lookupBean)
+  {
+    LookupExtractorFactoryContainer container = lookupBean.getContainer();
+    LOG.info("Starting lookup [%s]:[%s]", lookupBean.getName(), container);
+    try {
+      if (container.getLookupExtractorFactory().start()) {
+        LOG.info("Started lookup [%s]:[%s]", lookupBean.getName(), container);
+        return new AbstractMap.SimpleImmutableEntry<>(lookupBean.getName(), container);
+      } else {
+        LOG.error("Failed to start lookup [%s]:[%s]", lookupBean.getName(), container);
+        return null;
+      }
+    }
+    catch (Exception e) {
+      throw new RE(e, "Failed to start lookup [%s]:[%s]", lookupBean.getName(), container);
```

Comment:
How will InterruptedException from the future being cancelled be handled here?

Reply:
InterruptedException couldn't be thrown here. I narrowed the catched exception type here and at one more place